src/app/clusters/diagnostic-logs-server/BDXDiagnosticLogsProvider.cpp

```diff
@@ -230,14 +230,14 @@ void BDXDiagnosticLogsProvider::OnInternalError()
 {
     ChipLogError(BDX, "Internal Error");
     VerifyOrDo(mIsAcceptReceived, SendCommandResponse(StatusEnum::kDenied));
-    Reset();
+    mTransfer.AbortTransfer(GetBdxStatusCodeFromChipError(CHIP_ERROR_INTERNAL));
 }
 
 void BDXDiagnosticLogsProvider::OnTimeout()
 {
     ChipLogError(BDX, "Timeout");
     VerifyOrDo(mIsAcceptReceived, SendCommandResponse(StatusEnum::kDenied));
-    Reset();
+    mTransfer.AbortTransfer(GetBdxStatusCodeFromChipError(CHIP_ERROR_TIMEOUT));
```

Comment:
How does the transfer get reset after this point (and similarly in OnInternalError)? My understanding is AbortTransfer will trigger a status report to be sent and then the exchange will close since no response is expected for the status report. I don't think this class calls Reset when handling those (through OnMsgToSend or OnExchangeClosing), so would the rest of the internal state still be set?

Reply:
Yeah I guess you are right. I looked at BDX a long time ago. From what I remember for OTA we abort transfer and then waited sometime for the status report to go through and called Reset from what I recall. We could something similar. That being said I am not sure if the test is failing due to a genuine issue or merely not cleaning up after finishing the tests and overlaps the next test. I am looking at that now. will update the PR accordingly